@here/harp-datasource-protocol/lib/Theme.ts

```diff
@@ -801,121 +803,106 @@ export type Attr<T> = { [P in keyof T]?: T[P] | JsonExpr };
  *
  * @see [[PointTechniqueParams]].
  */
-export interface SquaresStyle extends BaseStyle {
+export interface SquaresStyle extends BaseStyle<PointTechniqueParams> {
     technique: "squares";
-    attr?: Attr<PointTechniqueParams>;
 }
 
 /**
  * Render feature as set of circles rendered in screen space.
  *
  * @see [[PointTechniqueParams]].
  */
-export interface CirclesStyle extends BaseStyle {
+export interface CirclesStyle extends BaseStyle<PointTechniqueParams> {
     technique: "circles";
-    attr?: Attr<PointTechniqueParams>;
 }
 
 /**
  * Render feature as POIs (icons and text) rendered in screen space.
  *
  * @see [[MarkerTechniqueParams]].
  */
-export interface PoiStyle extends BaseStyle {
+export interface PoiStyle extends BaseStyle<MarkerTechniqueParams> {
     technique: "labeled-icon";
-    attr?: Attr<MarkerTechniqueParams>;
 }
 
 /**
  * Render feature as line markers, which is a recurring marker along a line (usually road).
  *
  * @see [[MarkerTechniqueParams]].
  */
-export interface LineMarkerStyle extends BaseStyle {
+export interface LineMarkerStyle extends BaseStyle<MarkerTechniqueParams> {
     technique: "line-marker";
-    attr?: Attr<MarkerTechniqueParams>;
 }
 
 /**
  * Render feature as line.
  */
-export interface LineStyle extends BaseStyle {
+export interface LineStyle extends BaseStyle<MarkerTechniqueParams> {
     technique: "line";
     secondaryRenderOrder?: number;
     secondaryCategory?: string;
-    attr?: Attr<MarkerTechniqueParams>;
 }
 
 /**
  * Render feature as segments.
  */
-export interface SegmentsStyle extends BaseStyle {
+export interface SegmentsStyle extends BaseStyle<SegmentsTechniqueParams> {
     technique: "segments";
-    attr?: Attr<SegmentsTechniqueParams>;
 }
 
-export interface SolidLineStyle extends BaseStyle {
+export interface SolidLineStyle extends BaseStyle<SolidLineTechniqueParams> {
     technique: "solid-line" | "dashed-line";
     secondaryRenderOrder?: number;
```

Comment:
Remove secondary* it's already in SolidLineTechnqueParams ...

Reply:
done